tests/trackers/test_mlflow_tracker.py

```diff
@@ -0,0 +1,130 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SPDX-License-Identifier: Apache-2.0
+# https://spdx.dev/learn/handling-license-info/
+
+# Standard
+import copy
+import json
+import os
+import tempfile
+
+# Third Party
+from transformers.utils.import_utils import _is_package_available
+import pytest
+
+# First Party
+from tests.test_sft_trainer import (
+    DATA_ARGS,
+    MODEL_ARGS,
+    TRAIN_ARGS,
+    _get_checkpoint_path,
+    _test_run_inference,
+    _validate_training,
+)
+
+# Local
+from tuning import sft_trainer
+from tuning.config.tracker_configs import MLflowConfig, TrackerConfigFactory
+
+mlflow_not_available = not _is_package_available("mlflow")
+
+
+@pytest.mark.skipif(mlflow_not_available, reason="Requires mlflow to be installed")
+def test_run_with_mlflow_tracker_name_but_no_args():
+    """Ensure that train() raises error with mlflow tracker name but no args"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+
+        train_args.trackers = ["mlflow"]
+
+        with pytest.raises(
+            ValueError,
+            match="mlflow tracker requested but mlflow_uri is not specified.",
+        ):
+            sft_trainer.train(MODEL_ARGS, DATA_ARGS, train_args)
+
+
+@pytest.mark.skipif(mlflow_not_available, reason="Requires mlflow to be installed")
+def test_e2e_run_with_mlflow_tracker():
+    """Ensure that training succeeds with mlflow tracker"""
+
+    # mlflow performs a cleanup at callback close time which happens post the
+    # delete of this directory so we run into two issues
+    # 1. the temp directory cannot be cleared as it has open pointer by mlflow
+    # 2. mlflow complaints that it cannot find a run which it just created.
+    # this is a race condition which is fixed with mkdtemp() which doesn't delete
+    tempdir = tempfile.mkdtemp()
+
+    train_args = copy.deepcopy(TRAIN_ARGS)
+    train_args.output_dir = tempdir
+
+    # This should not mean file logger is not present.
+    # code will add it by default
+    # The below validate_training check will test for that too.
+    train_args.trackers = ["mlflow"]
+
+    tracker_configs = TrackerConfigFactory(
+        mlflow_config=MLflowConfig(
+            mlflow_experiment="unit_test",
+            mlflow_tracking_uri=os.path.join(tempdir, "mlflowdb.sqlite"),
```

Comment:
Nice. Should we add a another assert in the end of this test to verify the mlflow tracking worked? Something like:
```
assert os.path.exists(os.path.join(tempdir, "mlflowdb.sqlite") is True
```
and maybe that the content has the experiment name `unit_test`?

Reply:
I have added this check but the experiment name is under a custom data format by mlflow so hard to find so for now `mlflow` db is present should suffice for us as the run won't succeed without it.